### DISCUSSION TITLE
[Errno 104] Connection reset by peer, usually sooner rather than later

### DISCUSSION THREAD

**Discussion by greggo74** [ParadoxAlarmInterface/pai#406] [Q&A]
Recently discovered and installed PAI.   Worked straight up (well, almost) but I've found that it will sooner rather than later end up losing connection and fail to reconnect indefinitely, stuck in the loop below...

_2023-11-11 04:02:43,340 - ERROR    - PAI.paradox.connections.protocols - Connection was closed: [Errno 104] Connection reset by peer
2023-11-11 04:02:43,341 - ERROR    - PAI.paradox.connections.connection - Connection was lost
2023-11-11 04:02:44,102 - ERROR    - PAI - Connection to panel lost: Not connected. Restarting
2023-11-11 04:02:49,277 - ERROR    - PAI.paradox.connections.ip.connection - Unhandled exception while connecting (try 1/3): STUN Connect Request error: {'class': b'\x00\x00\x04', 'code': 47, 'reason': b'Connection Timeout or Failure\x00\x00\x00'}
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/paradox/connections/ip/connection.py", line 30, in connect
    await self._try_connect()
  File "/usr/local/lib/python3.9/site-packages/paradox/connections/ip/connection.py", line 147, in _try_connect
    await self.stun_session.connect()
  File "/usr/local/lib/python3.9/site-packages/paradox/connections/ip/stun_session.py", line 50, in connect
    stun_r = await self._stun_connect(xoraddr)
  File "/usr/local/lib/python3.9/site-packages/paradox/connections/ip/stun_session.py", line 73, in _stun_connect
    raise ConnectToSiteFailed(
paradox.exceptions.ConnectToSiteFailed: STUN Connect Request error: {'class': b'\x00\x00\x04', 'code': 47, 'reason': b'Connection Timeout or Failure\x00\x00\x00'}_

I'm running PAI in a docker on OMV (Debian) on the same node as my Mosquitto and HA dockers.  My SP7000 w/ IP150 module has a static IP on my router so there's no issues with that changing IP address.

A restart of PAI will reconnect straight away but again just disconnect and cycle reconnection attempts.

I don't want to raise an issue yet until I make sure it's not something I've done wrong in my setup.

Thoughts?

**Comment by yozik04** [comment]
I do not recommend using cloud approach. Use local connection instead.

**Reply by yozik04** [reply]
@greggo74 I have edited the code a bit. Try to use latest `dev`. Does it reconnect automatically after the change?

**Reply by greggo74** [reply]
Thanks.  I'm using docker but I'll work out how to update directly from GitHub and let you know. 
My network is pretty stable, though the connection to the unit is via Powerlines adaptor then into my router - not that should really affect things.

**Reply by greggo74** [reply]
I didn't work out how to update the version within my docker - docker o/s comes with no git to clone, etc. so it didn't seem like the quickest thing in the world to do quickly (so I'd have to learn how to create a docker image, etc. which I'm sure I can do but was a limit time-constrained today).
What I did try was to leave a ping from a local terminal running constantly to the IP of the system and it hasn't disconnected with a time-out for a couple of hours.  I wonder if the system itself is going into some sort of sleep mode if there's no events going on?

**Reply by greggo74** [reply]
Btw, I'm not using the cloud approach, I'm using the direct connection to my IP150 approach.

**Reply by yozik04** [reply]
According to the exception you have sent it connects to the cloud :)
Docker hub has development build. Use that. https://hub.docker.com/layers/paradoxalarminterface/pai/dev-latest/images/sha256-b0276c25c6d401b9993c5fbb1edff150c4c8a1be051f7b6f83645a5baf2945f4?context=repo

Please verify my change before switching to local connection.

**Reply by greggo74** [reply]
Ok, I see the issue.  I have both the IP address and the site stuff defined, but because the port wasn't uncommented it used all of the Swan stuff.
I think you could considered failing start-up if more than 1 method is configured, or make the start-up logs about config a little more verbose.

### IP Connection Details
IP_CONNECTION_HOST = '192.168.X.X'      # IP Module address when using direct IP Connection
#IP_CONNECTION_PORT = 10000                 # IP Module port when using direct IP Connection
IP_CONNECTION_PASSWORD = 'XXXX' # IP Module password. "paradox" is default.
IP_CONNECTION_SITEID = 'Home'          # SITE ID. IF defined, connection will be made through this method.
IP_CONNECTION_EMAIL = 'XXX@XXXX.XXX'            # Email registered in the site

If I uncomment the PORT, it just fails to connect completely.  
If I log onto the box via HTTP the IP150 Configuration link on the main page is disabled, so I cannot check what port has actually been configured - is there a way to find out or another default other than 10000?

From the logs:
2023-11-13 08:58:35,314 - INFO     - PAI.paradox.paradox - Using IP Connection
...
2023-11-13 08:58:40,519 - INFO     - PAI.paradox.connections.ip.commands - Authentication Success. IP(IP150) Module version 20, firmware: 4.30, serial: 71076979
...
2023-11-13 08:58:42,874 - INFO     - PAI.paradox.paradox - Panel Identified SP7000 version 4.94 build 0

I've just switched to your dev-latest, turned off my ping to the panel - which kept everything connected for 20hrs, and i'll see what happens
